schema/schema.go

```diff
@@ -348,36 +348,37 @@ func (schema *Schema) StateVersioning() bool {
 	return stateful
 }
 
-func (schema *Schema) SyncKeyTemplate() string {
+
+func (schema *Schema) SyncKeyTemplate() (syncKeyTemplate string, ok bool) {
 	syncKeyTemplateRaw, ok := schema.Metadata["sync_key_template"]
 	if !ok {
-		return ""
-	}
-	syncKeyTemplate, ok := syncKeyTemplateRaw.(string)
-	if !ok {
-		return ""
+		return
 	}
-	return syncKeyTemplate
+	syncKeyTemplate, ok = syncKeyTemplateRaw.(string)
+	return
 }
 
-func (schema *Schema) GenerateCustomPath(data map[string]interface{}) (string, error) {
-	tpl, err := pongo2.FromString(schema.SyncKeyTemplate())
-	if err != nil {
-		return "", err
+func (schema *Schema) GenerateCustomPath(data map[string]interface{}) (path string, err error) {
+	syncKeyTemplate, ok := schema.SyncKeyTemplate()
+	if !ok {
+		err = fmt.Errorf("Failed to read sync_key_template from schema %v", schema.URL)
+		return
 	}
-	path, err := tpl.Execute(pongo2.Context{}.Update(data))
+	tpl, err := pongo2.FromString(syncKeyTemplate)
 	if err != nil {
-		return "", err
+		return
```

Comment:
I'm not sure this returns the proper `err`. My understanding is that `tpl, err :=` creates a new `err` variable, but this `return` returns the one declared as a return argument.